cpp/src/gandiva/llvm_types.h

```diff
@@ -125,6 +147,20 @@ class LLVMTypes {
     return retval;
   }
 
+  enum DecimalMember {
+    kDecimalValueIdx,
+    kDecimalPrecisionIdx,
+    kDecimalScaleIdx,
+  };
+
+  llvm::StructType* decimal128_struct_type() {
+    return llvm::StructType::get(context_, {i128_type(), i32_type(), i32_type()}, false);
```

Comment:
This looks a bit wasteful. The types are always going to be the same along a single batch.

Reply:
This is a struct for a decimal128 _value_, right? Meaning you pass a separate scale and precision for each value?

Reply:
okay, I get what you are saying. I can separate out the scale/precision to a different structure, that is inited only once. 

I didn't realize this since the optimization passes inlined the entire add function, and removed the structs during my testing. but, I think it's better to not depend on that.

I'll fix this.